CMakeLists.txt

```diff
@@ -192,8 +198,12 @@ else()
 endif()
 
 if(NOT WIN32)
-  core_optional_dep(OpenGl)
-  if(OPENGL_FOUND AND NOT APPLE)
+  if(NOT ROCKCHIP_FOUND)
+    core_optional_dep(OpenGl)
+  else()
+    core_optional_dep(XRandR ENABLE_X11)
+  endif()
+  if(OPENGL_FOUND AND NOT APPLE AND NOT ROCKCHIP_FOUND)
```

Comment:
i agree: in general it needs a redesign to get out of this mess, enabling clear and straightforward platform based module selection. it is exactly what i bumped into while my goal primary is to give working hardware acceleration and rendering for current and coming rockchip devices back to the kodi and rockchip community.

is this redesign already scheduled and if so when and by who? i would be happy to help but platform display modules are not my expertise; rockchip hardware decoding and rendering are.